src/net.c

```diff
@@ -2830,6 +2830,11 @@ handle_request(coap_context_t *context, coap_session_t *session, coap_pdu_t *pdu
       respond = no_response(pdu, response, session);
       if (respond != RESPONSE_DROP) {
         coap_mid_t mid = pdu->mid;
+        if (COAP_RESPONSE_CLASS(response->code) != 2) {
```

Comment:
I wonder if we can integrate this with the following condional (code > 2).
Empty responses should not contain options, anyway, and we currently do not have 1.xx response codes.

(The same pattern would be applicable for the changes to resource.c.)

Reply:
I thought of doing that initially, but as there is no control over what the application updates the response code to, I thought it best to exactly follow the RFC (as libcoap now always adds in the Observe option before calling the request handler).  I can move it if you want.

Reply:
After having written down the alternative, I agree that we should leave it as it is; Otherwise, we would need to add special treatment for 1.xx response codes anyway because these would most likely not change, e.g., the observe relationship.